stan/math/prim/mat/fun/append_array.hpp

```diff
@@ -3,78 +3,216 @@
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
 #include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <vector>
+#include <iostream>
 
 namespace stan {
   namespace math {
+    /**
```

Comment:
Metaprograms should go into their own files in the `meta` directory.  That allows us to reuse them rather than continuously reinventing them in each application.